test/unit/stats.js

```diff
@@ -0,0 +1,793 @@
+        // assert.equal(err.propertyName, 'stat_id');
+        // assert.equal(err.propertyValue, req.params.stat_id);
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+
+    xit('should give an error if the stat_id is not an integer',
+    function(done) {
+      var req = {
+        params: {
+          stat_id: 'that stat'
+        }
+      };
+
+      var promise = stats.getStat(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given stat_id is of invalid format (e.g. not an integer or' +
+        ' negative)');
+
+        // assert.equal(err.name, 'InvalidArgumentError');
+        // assert.equal(err.propertyName, 'stat_id');
+        // assert.equal(err.propertyValue, req.params.stat_id);
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+
+    xit('should give an error if the stat_id is not in the database',
+    function(done) {
+      var req = {
+        params: {
+          stat_id: 9999999
+        }
+      };
+
+      var promise = stats.getStat(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given stat_id is not found in the database');
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+  });
+
+  describe('createStat(req)', function() {
+    // Post a new row of stats into stats database
+    xit('should add new stats to the database', function(done) {
+      var req = {
+        body: {
+          student_id: 2,
+          event_id: 2,
+          height: 320,
+          weight: 54,
+          pacer: 382,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStat(req);
+      })
+      .then(function() {
+        return stats.getStats({});
+      })
+      .then(function(data) {
+        assert.lengthOf(data, statCount + 1);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6, fakeStat7]);
+        done();
+      });
+    });
+
+    // TODO Insert test for trying to post if stat already exists
+    // (should do nothing. Make sure of that.)
+
+    // Attempt to post existing stat
+    xit('adding an existant set of stat should do nothing', function(done) {
+      var req = {
+        body: {
+          student_id: 1,
+          event_id: 1,
+          height: 5,
+          weight: 5,
+          pacer: 5
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStat(req);
+      })
+      .then(function() {
```

Comment:
change this .then to a .catch and check the error right here. The return stats.createStat line should return an error so the next block should be a .catch